python/lsst/ip/isr/linearize.py

```diff
@@ -679,6 +680,52 @@ def __call__(self, image, **kwargs):
             return False, 0
 
 
+class LinearizeSpline(LinearizeBase):
+    """Correct non-linearity with a spline model.
+
+    corrImage = uncorrImage - Spline(coeffs, uncorrImage)
+
+    Notes
+    -----
+
+    The spline fit calculates a correction as a function of the
+    expected linear flux term.  Because of this, the correction needs
+    to be subtracted from the observed flux.
+
+    """
+    LinearityType = "Spline"
+
+    def __call__(self, image, **kwargs):
+        """Correct for non-linearity.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.Image`
+            Image to be corrected
```

Comment:
This is obviously _waaaay_ beyond this scope of this review, but does this really only ever run on the `Image`? Doesn't the variance plane need to be subjected to (some function of) this correction too? Or is that so n-th order as to be deep in the weeds?

Reply:
That's the way it has been written for a long time.  However, it runs in `ip_isr` after the variance plane has been constructed, which makes me wonder if it should be being run prior to that step.  I think in general this is a small enough term to ignore that issue.  Since the nonlinearity is at the bright end, I think the astronomical effects of this would be making significant sources more significant, and possibly biasing surface brightness fits to galaxy cores.  I'll think about it more, as it's certainly out of scope here.